graylog2-server/src/main/java/org/graylog2/rest/resources/streams/StreamResource.java

```diff
@@ -325,6 +331,14 @@ public Response cloneStream(@ApiParam(name = "streamId", required = true) @PathP
             streamService.addAlertCondition(stream, alertCondition);
         }
 
+        for (AlarmCallbackConfiguration alarmCallbackConfiguration : alarmCallbackConfigurationService.getForStream(sourceStream)) {
```

Comment:
the same applies to the stream rules, i guess.
As long as we don't have transaction support, the best we could do is to collect the ids and types, wrap the entire thing in a try-catch block and remove the new entities on error.
Or simply punt ;)

For now my vote would be to rely on mongodb here and not go overboard with error handling.